2 dz/4/main.cpp

```diff
@@ -9,16 +9,16 @@ void val(int denominator)
 {
     double *fractionValues = new double[1000];
     int index = 0;
-    for (double den = 2; den <= denominator; den++)
+    for (int den = 2; den <= denominator; den++)
     {
-        for (double num = 1; num < den; num++)
+        for (int num = 1; num < den; num++)
         {
             int c = gcd(num, den);
             if (num < den && c == 1)
             {
                 fractions[index].numerator = num;
                 fractions[index].denominator = den;
-                fractions[index].arrValue = num / den;
+                fractions[index].arrValue = (double)num / den;
```

Comment:
We usually separate such constructions by a space, so `fractions[index].arrValue = (double) num / den;`